keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -159,7 +154,19 @@ def __init__(
         )
 
         # initialize trainable networks
-        self.feature_pyramid = feature_pyramid or layers_lib.FeaturePyramid()
+        if not feature_pyramid:
+            extractor_levels = [3, 4, 5]
+            extractor_layer_names = [
+                backbone.stack_level_outputs[i] for i in extractor_levels
+            ]
+            # TODO(bischof): consider calling this `self.feature_extractor`
+            self.backbone = backbone.get_feature_extractor(
+                extractor_layer_names, extractor_levels
+            )
+            self.feature_pyramid = layers_lib.FeaturePyramid()
+        else:
+            self.backbone = backbone
```

Comment:
This seems a bit inconsistent with the docstring.

It seems like, if `feature_pyramid` is specified, the `backbone` needs to be a keras.Model whose output feeds into the FPN as inputs.

Reply:
Yes, I wonder if we also need a `feature_extractor` argument to allow custom featurization? Or would having three components to specify (backbone, feature_extractor, and fpn) would get overwhelming for the user?

Reply:
@ianstenbit I think the reason for this change is that we don't want to expose featurization to the 99+% of users that just want to use our backbones. I'm hoping we can find a way to accomplish this.

Reply:
How about this contract:
* If `feature_pyramid` is `None`: `backbone` must implement `pyramid_level_outputs` property
* If `feature_pyramid` specified: `backbone` must implement `pyramid_outputs_extractor` method

Reply:
Maybe I'm missing something, but can't we just have the contract for the `backbone` be the same regardless of whether a `feature_pyramid` is specified?

The `backbone` always has to have `pyramid_level_outputs`, and the `feature_pyramid` has to accept a dictionary input with keys `3`, `4`, and `5`

Reply:
Good point! I was having difficulty anticipating how much flexibility users might want. If the only potential use case is building a pyramid that uses different levels, maybe this is enough

Reply:
Per offline discussion removing the custom FPN option and hardcoding the feature extraction inside the Task.